File: cpp/two-sum.cpp
// https://leetcode.com/problems/two-sum/

class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> x;

        x.reserve(nums.size());

        for (int i{}; i < nums.size(); ++i) {
            if (x.contains(target - nums[i])) {
                return {i, x[target - nums[i]]};
            } else {
                x.insert({nums[i], i});
            }
        }

        return {};
    }
};

File: java/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> x = new HashMap(nums.length);

        for (int i = 0; i < nums.length; ++i) {
            if (x.containsKey(target - nums[i])) {
                return new int[] {i, x.get(target - nums[i])};
            } else {
                x.put(nums[i], i);
            }
        }

        return new int[] {};
    }
}
